common/views/components/IIIFPresentationPreview/IIIFPresentationPreview.js

```diff
@@ -1,199 +1,156 @@
 // @flow
+import fetch from 'isomorphic-unfetch';
 import { iiifImageTemplate } from '@weco/common/utils/convert-image-uri';
-import { Fragment } from 'react';
-import styled from 'styled-components';
-import { classNames } from '../../../utils/classnames';
-import Button from '@weco/common/views/components/Buttons/Button/Button';
+import { Fragment, useEffect, useState } from 'react';
+import { type iiifPresentationLocation } from '@weco/common/utils/works';
 
-const BookContainer = styled.div`
-  display: inline-block;
-  position: relative;
-  z-index: 1;
-  margin-bottom: 20px;
-  word-spacing: 0;
+// Ideal preview thumbnails order: Title page, Front Cover, first page of Table of Contents, 2 random.
+// If we don't have any of the sructured pages, we fill with random ones, so there are always 5 images if possible.
+function randomImages(iiifManifest = null, structuredImages = [], n = 1) {
+  const images = [];
+  const canvases = iiifManifest
+    ? iiifManifest.sequences
+        .find(sequence => sequence['@type'] === 'sc:Sequence')
+        .canvases.filter(canvas => {
+          // Don't include the structured pages we're using when getting random ones
+          return (
+            structuredImages
+              .map(type => {
+                return type.images.find(image => {
+                  return canvas['@id'] === image.canvasId;
+                });
+              })
+              .filter(Boolean).length === 0
+          );
+        })
+    : [];
 
-  & > img,
-  &::before,
-  &::after {
-    /* Add shadow to distinguish sheets from one another */
-    box-shadow: 2px 1px 1px rgba(0, 0, 0, 0.15);
-  }
-
-  &::before,
-  &::after {
-    content: '';
-    position: absolute;
-    width: 100%;
-    height: 100%;
-    background-color: #fff;
-  }
-
-  /* Second sheet of paper */
-  &::before {
-    left: 7px;
-    top: 7px;
-    z-index: -1;
-  }
-
-  /* Third sheet of paper */
-  &::after {
-    left: 14px;
-    top: 14px;
-    z-index: -2;
-  }
+  const numberOfImages = canvases.length < n ? canvases.length : n;
 
-  img {
-    display: block;
+  for (var i = 1; i <= numberOfImages; i++) {
```

Comment:
Why not map instead of push?

Reply:
do you mean `map` instead of `for`? And why not `concat` instead of `push`?

Reply:
instead of using `const arr = []; arr.push(thing)` use `const arr = things.map(thing => thing)`.

Reply:
Not using map because I don't want to map the whole array. It's creating an array of n things (depending on how many random images are required).